src/benchmarks/reconstruction/lfhcal_studies/lfhcal_studiesProcessor.cc

```diff
@@ -313,15 +312,15 @@ void lfhcal_studiesProcessor::Process(const std::shared_ptr<const JEvent>& event
   int nCaloHitsSim = 0;
   float sumActiveCaloEnergy = 0;
   float sumPassiveCaloEnergy = 0;
-  auto simHits = event -> Get<edm4hep::SimCalorimeterHit>(nameSimHits.data());
-  for (const auto *caloHit : simHits) {
-    float x         = caloHit->getPosition().x / 10.;
-    float y         = caloHit->getPosition().y / 10.;
-    float z         = caloHit->getPosition().z / 10.;
-    uint64_t cellID = caloHit->getCellID();
-    float energy    = caloHit->getEnergy();
+  const auto &simHits = *static_cast<const edm4hep::SimCalorimeterHitCollection*>(event->GetCollectionBase(nameSimHits));
+  for (const auto caloHit : simHits) {
+    float x         = caloHit.getPosition().x / 10.;
+    float y         = caloHit.getPosition().y / 10.;
+    float z         = caloHit.getPosition().z / 10.;
+    uint64_t cellID = caloHit.getCellID();
+    float energy    = caloHit.getEnergy();
     double time = std::numeric_limits<double>::max();
-    for (const auto& c : caloHit->getContributions()) {
+    for (const auto c : caloHit.getContributions()) {
```

Comment:
Maybe -Wpedantic but should we keep the reference?
```suggestion
    for (const auto& c : caloHit.getContributions()) {
```
(applies in other places too)

Reply:
Good catch! Although, I only see one place where this needed to be fixed.